android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -791,8 +791,23 @@ public void clearCookies(String url)
 	public void setRequestHeader(String header, String value)
 	{
 		if (readyState <= READY_STATE_OPENED) {
-			headers.put(header, value);
-
+			if (value == null) {
+				// If value is null, remove header
+				headers.remove(header);
```

Comment:
The idea was, if you set cookies, it will be appended and if you set null, it is cleared. A null value is being used as a way to clear all the values since we only have a set method.

Should this behavior be changed?